### DISCUSSION TITLE
`texdoc` cannot find any documentation

### DISCUSSION THREAD

**Discussion by liblaf** [Homebrew/discussions#3604] [Linux] [answered]
I want to use `texdoc` to find documentation for packages, but got the following output.

```console
$ texdoc texdoc
Sorry, no documentation found for "texdoc".
If you are unsure about the name, try full-text searching on CTAN.
Search form: <https://www.ctan.org/search/>
```

How can I fix this?

**Comment by carlocab** [comment]
How did you install `texdoc`? Is that from the `texlive` formula, or a `mactex` cask?

**Reply by liblaf** [reply]
From the `texlive` formula.

**Comment by danielnachun** [comment, marked as answer]
If you used the `texlive` formula, we had to delete the documentation because it was huge and the bottles is already massive as is.  Generally speaking we do not ship documentation other than man pages if it takes up too much space or requires a lot of extra dependencies to build.  PDF and HTML documentation should be available online and if it is not, I would consider that an issue the upstream project needs to fix.

**Reply by liblaf** [reply]
Thanks a lot.
I also found that <https://texdoc.org> is a good choice for online documentation.

**Reply by carlocab** [reply]
We can probably add the documentation as a separate formula. Tracking down `tex` documentation is tricky, and we probably want the ones that match the version we ship.

**Reply by liblaf** [reply]
> We can probably add the documentation as a separate formula. Tracking down `tex` documentation is tricky, and we probably want the ones that match the version we ship.

Good idea! Arch Linux provides a package called [texlive-most-doc](https://aur.archlinux.org/packages/texlive-most-doc). I hope a similar formula may also be added to brew.

**Comment by ammantesfaye** [comment]
Is there any way to install the docs as a separate formula?

**Reply by liblaf** [reply]
Not yet. But you can use online documentation like [texdoc.org](https://texdoc.org). For example, add the function to `.bashrc` or `.zshrc`:

```sh
function texdoc() {
  open "https://texdoc.org/pkg/${1}" > /dev/null 2>&1
  # or xdg-open ...
}
```

It can be used like normal `texdoc`.
